lua/rust-tools/hover_actions.lua

```diff
@@ -63,73 +50,29 @@ function M.handler(_, result, ctx)
     return
   end
 
-  local markdown_lines =
-    util.convert_input_to_markdown_lines(result.contents, {})
-  if result.actions then
-    M._state.commands = result.actions[1].commands
-    local prompt = parse_commands()
-    local l = {}
-
-    for _, value in ipairs(prompt) do
-      table.insert(l, value)
-    end
-
-    markdown_lines = vim.list_extend(l, markdown_lines)
-  end
-
-  markdown_lines = util.trim_empty_lines(markdown_lines)
-
-  if vim.tbl_isempty(markdown_lines) then
-    -- return { 'No information available' }
+  if
+    not (
+      rt.config
+      and rt.config.options
+      and rt.config.options.tools
+      and rt.config.options.tools.hover_actions
+      and rt.config.options.tools.hover_actions.handler
+    )
+  then
+    -- return { 'No handler configured' }
     return
   end
 
-  local bufnr, winnr = util.open_floating_preview(
-    markdown_lines,
-    "markdown",
-    vim.tbl_extend("keep", rt.config.options.tools.hover_actions, {
-      focusable = true,
-      focus_id = "rust-tools-hover-actions",
-      close_events = { "CursorMoved", "BufHidden", "InsertCharPre" },
-    })
-  )
-
-  if rt.config.options.tools.hover_actions.auto_focus then
-    vim.api.nvim_set_current_win(winnr)
-  end
+  local options = {
+    contents = result.contents,
+  }
 
-  if M._state.winnr ~= nil then
-    return
-  end
-
-  -- update the window number here so that we can map escape to close even
-  -- when there are no actions, update the rest of the state later
-  M._state.winnr = winnr
-  vim.keymap.set(
-    "n",
-    "<Esc>",
-    close_hover,
-    { buffer = bufnr, noremap = true, silent = true }
-  )
-
-  vim.api.nvim_buf_attach(bufnr, false, {
-    on_detach = function()
-      M._state.winnr = nil
-    end,
-  })
-
-  --- stop here if there are no possible actions
-  if result.actions == nil then
-    return
+  if result.actions then
+    options.actions = parse_commands(result.actions[1].commands, ctx)
   end
 
-  -- makes more sense in a dropdown-ish ui
-  vim.api.nvim_win_set_option(winnr, "cursorline", true)
-
-  -- run the command under the cursor
-  vim.keymap.set("n", "<CR>", function()
-    run_command(ctx)
-  end, { buffer = bufnr, noremap = true, silent = true })
+  local handler = require(rt.config.options.tools.hover_actions.handler)
+  handler(options)
```

Comment:
Instead of the dynamic `require`, I modified here just `rt.config.options.tools.hover_actions.handler()`. I was a little confused to switch to the builtin handler with this dynamic require approach.